doc/ref_arch/kubernetes/chapters/chapter01.rst

```diff
@@ -0,0 +1,115 @@
+Overview
+========
+
+Introduction
+------------
+
+The intention of this Reference Architecture is to develop a usable Kubernetes based platform for the Telecom operator community. The RA will be based on the standard Kubernetes platform where ever possible. This Reference Architecture for Kubernetes will describe the high level system components and their interactions, taking the `goals and requirements <../../../common/chapter00.md>`__ and mapping them to real-world Kubernetes (and related) components. This document needs to be sufficiently detailed and robust such that it can be used to guide the production deployment of Kubernetes within an operator, whilst being flexible enough to evolve with and remain aligned with the wider Kubernetes ecosystem outside of Telco.
```

Comment:
all references to CNTT md documents should be changed to rst else changes made in rst files going forward wouldn't be refelcted in the links

Reply:
At the moment we use a Sphinx plugin to blindly change all ".md"-s to ".html"-s in the relative links. Becouse of this all links are automagically working, however this is a bit of hacking of how rst works. I started to correct this and then realized that with `:doc:` directives the # anchors do not work, so the only way to refer correctly to chapters is to use `:ref:`-s what would require a manual annotating of all headings. Somehow I feel that at least for the beginning we should just keep using the relative-link-corrector for the existing links and start using proper rst linking for new text. Gradually we can start replacing all ".md" links with proper ones later. @pgoyal01 , @collivier , @gkunz , @karinesevilla  what do you think?

Reply:
@CsatariGergely That would be fine but if edit the referenced equivalent rst file then we would need to change references to the md version everywhere else we would be pointing to outdated content. That can be error-prone. 

It may be worth dividing up the work amongst us -- assign Issues (the WSLs can do that) -- to go and add `:ref:`-s to each section header. We need to use a standard format and use only the text - no number. Wherever a reference uses a # should be changed to one with corresponding text.

Other Ideas?

Reply:
Somehow I also have a feeling that manual labels are the most optimal on the long run, but I would like to get the conversion pr-s get merged before we start addressing them.

Reply:
Cross-posting from another thread: In the RI1 and RI2 conversions, I have been using the :ref: directive in conjunction with the autosectionlabel extension to automatically create header references:

For intra-references using implicit header references, see https://github.com/cntt-n/CNTT/pull/2777/files#diff-774fcc11722dce98d0347ce9870d8c0959a06c61223d2ee4a1cc4516db7211b5R64

You have to use the autosectionlabel extention (see https://github.com/cntt-n/CNTT/pull/2777/files#diff-e59005d178179a798a25d32c0f18a66b77b4cd290462ac82d5594c23fb0ee0d6R15)

See also these options: https://github.com/cntt-n/CNTT/pull/2777/files#diff-e59005d178179a798a25d32c0f18a66b77b4cd290462ac82d5594c23fb0ee0d6R62 to avoid clashes of header references across files.

Reply:
Thanks for the references. I've tried the approach, but for some reason it is not working for me. I would propose to merge the big rst conversion pr-s and fix smaller issues in different pr-s. Smaller issues so far are:
  - Consistent mechanism for referencing sections (preferably with :ref: and autosectionlabel)
  - Support for top level and document level builds